### PR TITLE
Update mail_message.py

### PR DESCRIPTION
**Pull request by KomarovAlea**
Two identical conditions in L288 and L290 
The condition needs to be removed L332.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by KomarovAlea** [comment]
What's wrong with my PR?

**Comment by pedrobaeza** [comment]
As said, the CLA, but this PR has more problems:

* You have to explain what are you trying to fix or improve.
* Why this is better?
* Give a proper title

**Comment by mart-e** [comment]
Waiting for #21399 for the CLA but, as said Pedro, explain *why* you are doing the changes, not *what*.

File: addons/mail/models/mail_message.py [hunk 1]
```diff
@@ -328,12 +325,7 @@ def _message_read_dict_postprocess(self, messages, message_tree):
                 author = partner_tree[message.author_id.id]
             else:
                 author = (0, message.email_from)
-            partner_ids = []
-            if message.subtype_id:
-                partner_ids = [partner_tree[partner.id] for partner in message.partner_ids
-                                if partner.id in partner_tree]
-            else:
-                partner_ids = [partner_tree[partner.id] for partner in message.partner_ids
+            partner_ids = [partner_tree[partner.id] for partner in message.partner_ids
```
Comment (arkostyuk) [inline addons/mail/models/mail_message.py]: Well, this one is correct.

File: addons/mail/models/mail_message.py [hunk 2]
```diff
@@ -285,10 +285,7 @@ def _message_read_dict_postprocess(self, messages, message_tree):
         for message in message_tree.values():
             if message.author_id:
                 partners |= message.author_id
-            if message.subtype_id and message.partner_ids:  # take notified people of message with a subtype
-                partners |= message.partner_ids
-            elif not message.subtype_id and message.partner_ids:  # take specified people of message without a subtype (log)
-                partners |= message.partner_ids
+            partners |= message.partner_ids
```
Comment (arkostyuk) [inline addons/mail/models/mail_message.py]: Not cool 🙂 

The original `if-elif` block was kinda weird, yeah, but that condition still took `message.partner_ids` into account
I don't really think that removing the whole condition is a good idea, and that's why:

With 
```python
x = message.subtype_id
y = message.partner_ids
```
and using `foo()` as an alias for `partners |= message.partner_ids`, this roughly translates to:
```python
if x and y:
    foo()
elif not x and y:
    foo()
```
Furthermore:
```python
if (x and y) or (not x and y):
    foo()
```
At this point, this is safe to throw `x` expression part out of the condition:
```python
if y:
    foo()
```
Just translate it back to original variables, we'll get
```python
if message.partner_ids:
    partners |= message.partner_ids
```
...but the condition is still there. Keep it pls. 🙂

**Comment by tde-banana-odoo** [comment]
Hello,

I still don't understand the purpose of this PR. Indeed code could be improved but it does not seem to have any functional or performance improvements. As we plan to rewrite part of that code I don't think this PR adds value to the current code. Could you maybe explain more in details the purpose of your code ?

Thanks,

**Comment by KomarovAlea** [comment]
Hello,

My code does not really add functionality and performance. I suggested this change, because this piece of code is strange and everyone recognizes it. The fact is that this oddity arose as a result of the gradual evolution of the functionality of this (very important) module. If you look into the history of the evolution of this module, you will see that there **was** a relations between the parameters: "subtype_id" and "partner_ids", which was lost. I did not go into details: "Is this relations now necessary" or "there is no longer any need for it." But the very fact of the existence of such a strange code, suggests that the previous modifications were not fully implemented. And this, in turn, indicates that there may be a mistake or vulnerability in this piece of code. I want to emphasize: "there **can** be a vulnerability." I do **not** say that she is there. And that I and other developers did not rack their brains and did not waste time searching for an answer to this question - I created this PR. Those. in order to put an end to this question. Yes, I'm a perfectionist. Yes, this code can not be touched and left as is. 
If you wish, you can close this PR.

**Comment by tde-banana-odoo** [comment]
Hello,

Thanks for explaining your point of view. In master this code will be changed to better reflect current state of business flow and will be easier to understand. As there is no bug in this code currently we won't change it to avoid adding regression. I understand your point of view while our policy is more strict. I therefore close this PR.

Many thanks,

**Pull request closed without merging**